plugins/c8y_configuration_plugin/src/upload.rs

```diff
@@ -89,19 +123,216 @@ pub async fn handle_config_upload_request(
     Ok(())
 }
 
-async fn upload_config_file(
+pub async fn upload_config_file(
     config_file_path: &Path,
     config_type: &str,
     http_client: &mut impl C8YHttpProxy,
 ) -> Result<String> {
     // upload config file
     let upload_event_url = http_client
-        .upload_config_file(config_file_path, config_type)
+        .upload_config_file(config_file_path, config_type, None)
         .await?;
 
     Ok(upload_event_url)
 }
 
+/// Map the c8y_UploadConfigFile request into a tedge/commands/req/config_snapshot command for the child device.
+/// The child device is expected to upload the config fie snapshot to the file transfer service.
+/// A unique URL path for this config file, from the file transfer service, is shared with the child device in the command.
+/// The child device can use this URL to upload the config file snapshot to the file transfer service.
+pub async fn handle_config_upload_request_child_device(
+    config_upload_request: SmartRestConfigUploadRequest,
+    mqtt_client: &mut Connection,
+    local_http_host: &str,
+) -> Result<()> {
+    let child_id = config_upload_request.device;
+    let config_type = config_upload_request.config_type;
+
+    let plugin_config = PluginConfig::new(Path::new(&format!(
+        "/etc/tedge/c8y/{child_id}/c8y-configuration-plugin.toml"
+    )));
+
+    let file_entry = plugin_config.get_file_entry_from_type(&config_type)?;
+
+    let config_management = ConfigOperationRequest::Snapshot {
+        child_id,
+        file_entry,
+    };
+
+    info!("Sending config snapshot request to child device");
+    let msg = Message::new(
+        &config_management.operation_request_topic(),
+        config_management.operation_request_payload(local_http_host)?,
+    );
+    mqtt_client.published.send(msg).await?;
+
+    Ok(())
+}
+
+pub async fn handle_child_device_config_snapshot_response(
+    message: &Message,
+    config_management: &ConfigOperationRequest,
```

Comment:
Surprised to see a `ConfigOperationRequest` as an argument to this function that handles an operation **response**. I'd have expected it to take a `ConfigOperationResponse` instead. Same for the function `handle_child_device_successful_config_snapshot_response` below.